3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala

```diff
@@ -311,53 +312,37 @@ object ConversionUtils {
-   * 2. IllegalArgumentException
-   *    if uri violates RFC 2396
-   *
-   * @param uri The URI string to
-   *        convert
-   * @param useNetaporter Whether to use the
-   *        com.netaporter.uri library
-   * @return an Option-boxed URI object, or an
-   *         error message, all
-   *         wrapped in a Validation
+   * Parses a string to create a [[URI]].
+   * Parsing is relaxed, i.e. even if a URL is not correctly percent-encoded or not RFC 3986-compliant, it can be parsed.
+   *
+   * @param uri String containing the URI to parse.
+   * @return [[Validation]] wrapping the result of the parsing:
+   *         - [[Success]] with the parsed URI if there was no error or with [[None]] if the input was `null`.
+   *         - [[Failure]] with the error message if something went wrong.
    */
-  def stringToUri(uri: String, useNetaporter: Boolean = false): Validation[String, Option[URI]] =
-    try {
-      val r = uri.replaceAll(" ", "%20") // Because so many raw URIs are bad, #346
-      Some(URI.create(r)).success
-    } catch {
-      case npe: NullPointerException => None.success
-      case iae: IllegalArgumentException =>
-        useNetaporter match {
-          case false => {
-            val netaporterUri = try {
-              Uri.parse(uri).success
-            } catch {
-              case NonFatal(e) =>
-                "Provided URI string [%s] could not be parsed by Netaporter: [%s]"
-                  .format(uri, ExceptionUtils.getRootCause(iae).getMessage)
-                  .fail
-            }
-            for {
-              parsedUri <- netaporterUri
-              finalUri  <- stringToUri(parsedUri.toString, true)
-            } yield finalUri
-          }
-          case true =>
-            "Provided URI string [%s] violates RFC 2396: [%s]"
-              .format(uri, ExceptionUtils.getRootCause(iae).getMessage)
-              .fail
-        }
-      case NonFatal(e) => "Unexpected error creating URI from string [%s]: [%s]".format(uri, e.getMessage).fail
-    }
+  def stringToUri(uri: String): Validation[String, Option[URI]] =
+    Try(
+      Option(uri) // to handle null
+        .map(_.replaceAll(" ", "%20"))
+        .map(URI.create)
+        .success
+    ).recoverWith {
+      case NonFatal(javaErr) =>
+        implicit val c =
+          UriConfig(decoder = PercentDecoder(ignoreInvalidPercentEncoding = true), encoder = percentEncode -- '+')
+        Uri
+          .parseTry(uri)
+          .map(_.toJavaURI)
+          .transform( // creates Try[Validation[String, Option[URI]]]. Try will always be a Success
+            s => util.Success(Some(s).success),
+            scalaErr =>
+              util.Success(
+                "Provided URI [%s] could not be parsed, neither by Java parsing (error: [%s]) nor by Scala parsing (error: [%s])."
+                  .format(uri, javaErr.getMessage, scalaErr.getMessage)
+                  .failure
+            )
+          )
+    }.get
```

Comment:
avoid `get`

Reply:
I was 100% sure that it was a `Success`. I replaced by pattern matching